### PR TITLE
Use relative paths in redirects

### PR DESCRIPTION
**Pull request by nsklikas**
When the UI is served behind a reverse proxy (e.g. traefik), the pages may be served under a subpath. This breaks our redirects and calls to the backend API. In order to overcome this we use relative paths everywhere.

This is not the proper solution and ideally we would want the reverse proxy to inject the html with the base URL tag.

### CODE REVIEW

File: ui/pages/login.tsx [hunk 1]
```diff
@@ -102,12 +105,12 @@ const Login: NextPage = () => {
...
-          <Card title="Choose Provider" >
-            <Flow onSubmit={onSubmit} flow={flow} />
-          </Card>:
-          <Spinner></Spinner>
-          }
+            {flow ?
+              <Card title="Choose Provider" >
+                <Flow onSubmit={onSubmit} flow={flow} />
+              </Card> :
+              <Spinner></Spinner>
```
Comment (lorumic) [inline ui/pages/login.tsx:112]: We don't have any content inside this component, so we can use a self-closing tag here:
```suggestion
              <Spinner />
```

**Review by lorumic — COMMENTED** [review]
I cannot help with the traefik part as I don't really know much about it, but maybe @edlerd has an opinion about this relative path issue?

Apart from that, there is only one small "code style" suggestion that I'm adding below.

**Review by edlerd — DISMISSED** [review]
LGTM
Have upstream provided a `BASE_URL` via config (or environment) might be safer for the future.

**Review by natalian98 — DISMISSED** [review]

**Review by bencekov — APPROVED** [review]
Looks good!

**Pull request merged**